libvast/vast/default_table_slice.hpp

```diff
@@ -63,4 +42,24 @@ class default_table_slice::builder : public table_slice::builder {
   default_table_slice_ptr slice_;
 };
 
+} // namespace detail
+
+/// The default implementation of `table_slice`.
+class default_table_slice : public table_slice {
+  friend detail::default_table_slice_builder;
+
+public:
+  /// Constructs a builder that generates a default_table_slice.
+  /// @param layout The layout of the table_slice.
+  /// @returns The builder instance.
+  static table_slice_builder_ptr make_builder(record_type layout);
+
+  default_table_slice(record_type layout);
+
+  virtual caf::optional<data_view> at(size_type row, size_type col) const final;
+
+private:
+  std::vector<data> xs_;
```

Comment:
Why not simply `vector` (`vast::vector` is defined as `std::vector<vast::data>`)?

Reply:
Sure, this would work as well. But the reason why I chose `std::vector<data>` is to emphasize that the table slice is a (standard) sequence of `data`. The semantics of `vector` typically imply that we're dealing with one data instance that happens to be a container.